packages/common-all/src/vault.ts

```diff
@@ -66,6 +66,11 @@ export class VaultUtils {
     return vault.fsPath;
   }
 
+  static getRelVaultRootPath(vault: DVault) {
+    if (VaultUtils.isSelfContained(vault)) return vault.fsPath;
+    return VaultUtils.getRelPath(vault);
```

Comment:
any reason not to put this logic into `VaultUtils.getRelPath()` itself? It's maybe a little bit confusing on knowing when to use `VaultUtils.getRelVaultRootPath()` vs `VaultUtils.getRelPath()`

Reply:
`getRelPath` gets the location of the notes, while `getRelVaultRootPath` gets the root of the vault. This is only different for self contained vaults, where `getRelPath() == getRelVaultRootPath() + "/notes"`.

When first implementing self contained vaults, I added a feature into `getRelPath` to get the path of the notes folder, since most of the code uses it to access the note files and schema files. But there are a few parts that need the root of the vault, so I'm adding this.

One option would be to rename `getRelPath` to `getRelNotesPath`, and `getRelVaultRootPath` can be the new `getRelPath`. I avoided doing that since it would be a very large diff, but I'm happy to do so.

Reply:
Maybe 80/20 is to just add some of this info as docstrings to these two util methods.

Reply:
Good idea, I'll add the docstrings and merge.